llvm/lib/IR/AutoUpgrade.cpp

```diff
@@ -1072,6 +1073,14 @@ static bool upgradeIntrinsicFunction1(Function *F, Function *&NewFn,
             {F->getReturnType(), F->getArg(1)->getType()});
         return true;
       }
+      // Old-style make.buffer.rsrc was only variadic in the input pointer
+      if (Name.consume_front("make.buffer.rsrc.") && Name.size() == 2) {
```

Comment:
This is assuming it's p + single digit? It's better to rely on the actual signature used

For this case I'm not sure you actually need any code to handle the upgrade. Does it work if you just delete this?

Reply:
Will take a look

But if anything, I think the alternate test is if there'll be one `.` or two.

Reply:
Yeah, turns out remangling for something becoming more variadic is automatic